src/interfaces/swig/SGBase.i

```diff
@@ -364,7 +364,14 @@ INTERFACE_BASETYPE get(const std::string& name) const
         }
 
         auto visitor = VISITOR_NAME(result);
-        find_iter->second->get_value().visit(&visitor);
+        try {
+            find_iter->second->get_value().visit(&visitor);
+        }
+        catch (const std::bad_optional_access&) {
+            error("The value of parameter {}::{} is automatically inferred during model fitting, "
+                  "and is currently not set. Either set a value with put or call get after model fitting.",
+                  $self->get_name(), name);
```

Comment:
@karlnapf is this error message fine?

Reply:
yes it is good. Maybe some nitpicking:

* don't mention method names (put), but just say "set a value", "read value after fitting"
* fit -> train (more shogun-ish)
* inferred -> is automatically computed using HEURISTIC_NAME ... (if the name is accessible at this stage)